client/src/components/charts/LegendsSortable.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { SortableContainer, sortableElement, sortableHandle } from 'react-sortable-hoc';
+import isEqual from 'lodash/isEqual';
+import get from 'lodash/get';
+import arrayMove from 'array-move';
+
+import { sortAlphabetically, sortChronologically } from '../../utilities/utils';
+import { palette } from '../../utilities/visualisationColors';
+import LegendShape from './LegendShape';
+
+const sortLegendsFunctionFactory = visualisation => (get(visualisation, 'data.common.metadata.type') === 'text' ?
+  sortAlphabetically : sortChronologically);
+
+const sortableItemStyle = { display: 'flex', alignItems: 'center', flexDirection: 'row', margin: '5px 0px 5px 5px' };
+
+const NoSortableItem = ({ value, color }) =>
+(<div style={sortableItemStyle} key={`item-${value}`} >
+  <LegendShape fill={color} />
+  <div style={{ marginLeft: '4px' }}>{value}</div>
+</div>);
+
+NoSortableItem.propTypes = {
+  value: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+};
+
+const DragHandle = sortableHandle(() => <div style={{ marginRight: '4px' }}><span>::</span></div>);
+
+const SortableItem = sortableElement(({ index, value, color }) =>
+   (<div style={sortableItemStyle}>
+     <DragHandle index={index} key={value} />
+     <LegendShape fill={color} />
+     <div style={{ marginLeft: '4px' }}>{value}</div></div>));
+
+SortableItem.propTypes = {
+  value: PropTypes.string.isRequired,
+  index: PropTypes.number.isRequired,
+};
+
+const getLegends = (specLegend, visualisation) => {
+  const specLegendsList = get(specLegend, 'order.list');
+
+  const visLegendsList = get(visualisation, 'data.common.data') || [];
+
+  return isEqual(new Set(specLegendsList), new Set(visLegendsList.map(l => l.key))) ?
+  specLegendsList : visLegendsList.map(l => l.key).sort(sortLegendsFunctionFactory(visualisation));
+};
+
+// ensure spec legend has order object
+export const ensureSpecLegend = (specLegend) => {
+  const legend = { ...specLegend } || {};
+  const order = { ...legend.order } || {};
+  legend.order = order;
+  return legend;
+};
+
+export const LegendsSortable = ({
+  onChangeSpec,
+  colors,
+  specLegend,
+  visualisation,
+}) => {
+  const sortable = (get(specLegend, 'order.mode') || 'auto') !== 'auto';
```

Comment:
Why not the positive case here? i.e. 

```suggestion
  const sortable = (get(specLegend, 'order.mode') || 'auto') === 'custom';
```